### PR TITLE
[NF] Add Motion correction workflow

### PR DESCRIPTION
**Pull request by skoudoro**
This PR just adds explicitly a new function named `motion_correction` (Between-Volumes Motion correction). It also adds the associated workflow.  This function has been requested many times since the last Workshop.

To use it, you just to do the following import: `from dipy.align import motion_correction`

Todo:
------
- [x] Add tests
- [x] Add tutorial for motion_correction()
- [x] Add tutorial for the workflow

### CODE REVIEW

File: dipy/workflows/align.py [hunk 1]
```diff
@@ -704,3 +707,66 @@ def run(self, static_image_files, moving_image_files, prealign_file='',
...
+
+            # Load the data from the input files and store into objects.
+            logging.info('Loading {0}'.format(dwi))
+            data, affine = load_nifti(dwi)
+
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+            print(b0_threshold, bvals.min())
+            if b0_threshold < bvals.min():
+                warn("b0_threshold (value: {0}) is too low, increase your "
+                     "b0_threshold. It should be higher than the first "
```
Comment (arokem) [inline dipy/workflows/align.py:760]: Should be higher than the lowest?

File: dipy/workflows/align.py [hunk 2]
```diff
@@ -704,3 +707,66 @@ def run(self, static_image_files, moving_image_files, prealign_file='',
             save_nifti(owarped_file, warped_moving, static_grid2world)
             logging.info('Saving Diffeomorphic map {0}'.format(omap_file))
             save_nifti(omap_file, mapping_data, mapping.codomain_world2grid)
+
+
+class MotionCorrectionFlow(Workflow):
+    """
+    The Motion Correction workflow allows the user to use only one type of
+    registration (such as center of mass or rigid body registration only).
```
Comment (arokem) [inline dipy/workflows/align.py:715]: Not clear what this means

**Review by arokem — COMMENTED** [review]
Great work. I have a couple of small questions/comments.

File: dipy/workflows/align.py (hunk 1 above)
Comment (skoudoro) [reply dipy/workflows/align.py:760]: yes, sounds better the way you said it, I will update.

File: dipy/workflows/align.py (hunk 2 above)
Comment (skoudoro) [reply dipy/workflows/align.py:715]: Oops, I forgot to finish to update this description, good catch

File: dipy/align/_public.py [hunk 3]
```diff
@@ -670,6 +671,16 @@ def register_dwi_series(data, gtab, affine=None, b0_ref=0, pipeline=None):
     return nib.Nifti1Image(data_array, affine), affine_array
 
 
+motion_correction = partial(register_dwi_series, pipeline=["center_of_mass",
+                                                           "translation",
+                                                           "rigid", "affine"])
+motion_correction.__doc__ = re.sub('Register.*?volume', 'Apply a motion '
+                                   'correction to a DWI dataset '
+                                   '(Between-Volumes Motion correction',
```
Comment (arokem) [inline dipy/align/_public.py:679]:
```suggestion
                                   '(Between-Volumes Motion correction)',
```

**Review by arokem — COMMENTED** [review]
Looks great. One small typo fix proposed.

**Comment by skoudoro** [comment]
Any additional comments?

**Comment by arokem** [comment]
Are the CI failures unrelated to the changes in this PR? If so, I think this is ready to go!

**Comment by skoudoro** [comment]
I don't think so and the build is too old so I can't access the log to be 100% sure. I will need to push an empty commit.

**Comment by arokem** [comment]
Still failing. Did you rebase?

**Comment by skoudoro** [comment]
All green now! Ready to go @arokem!

**Pull request merged**